### DISCUSSION TITLE
Issue with custumizing kedro datasets and the Pokemon Tutorial

### DISCUSSION THREAD

**Discussion by rosanajurdi** [kedro-org/kedro#3003] [Idea]
Hello all,
I am struggling with creating custumized datasets in kedro. I am working on the pokemon example in the kedro tutorial handbook. I have created the following code but everytime I try to run, I arrive at the following error.
Does anyone know the origin of the error and how I can fix it ?
 Thank you in advance !
 
# catalog:

pikachu:
  type: pokemon.extras.datasets.image_dataset.ImageDataSet
  filepath: data/01_raw/pokemon-images-and-types/images/images/pikachu.png

pokemon:
  type: PartitionedDataSet
  dataset: pokemon.extras.datasets.image_dataset.ImageDataSet
  path: data/01_raw/pokemon-images-and-types/images/images
  filename_suffix: ".png"
# pipeline:

from kedro.pipeline import Pipeline, pipeline, node
from .nodes import load_pokemon_images

def create_pipeline():
    return Pipeline(
        [
            node(
                func=load_pokemon_images,
                inputs = ["pokemon"],
                outputs = "pokemon_images",
                tags="pokemon",  # Use the dataset name defined in your catalog
            ),
        ]
    )
# node:

def load_pokemon_images(dataset):
    # Load the dataset using your custom ImageDataSet
    images = dataset.load()
    
    # Optionally, you can process or manipulate the loaded data here
    # For example, if you want to return the number of images loaded:
    num_images = len(images)
    
    return images, num_images

# pipeline_registery:
def register_pipelines() -> dict[str, Pipeline]:
    """Register the project's pipelines.

    Returns:
        A mapping from pipeline names to ``Pipeline`` objects.
    """
    pipelines = find_pipelines()
    pipelines["__default__"] = sum(pipelines.values())
    return pipelines

**Comment by astrojuanlu** [comment]
Cross posted in Slack: https://linen-slack.kedro.org/t/15624349/hello-all-i-am-struggling-with-creating-custumized-datasets-#2bf79516-a34e-4885-83a5-52f7a8177fec (and following posts)
